virtcontainers/qemu_arm64.go

```diff
@@ -27,7 +23,9 @@ const defaultQemuMachineType = QemuVirt
 
 const qmpMigrationWaitTimeout = 10 * time.Second
 
-var defaultQemuMachineOptions = "usb=off,accel=kvm,gic-version=" + getGuestGICVersion()
+const defaultQemuMachineOptions = "usb=off,accel=kvm,gic-version=host"
```

Comment:
A couple of questions:

- What is the minimum version of qemu that supports this option?
- What is the minumum version of Linux kernel supporting querying of the GIC version using an ioctl?

I agree it makes sense to let qemu handle this iff we can guarantee both it and the host kernel can deal with it.

Reply:
Further more, we can't use similar kvm_arm_vgic_probe() in Kata-runtime because of internal qemu structure kvm_state in kvm_create_device(kvm_state,...).   
Thus, after we add "gic-version=host"
1.  the qemu knows the correct gic-version which should be created for the guest
2. Kata-runtime doesn't know the correct gic-version, hence MaxQemuVCPUs() can't set max vcpu by gic-version.

The workaround is in my 2nd patch. I used default gic-version as 3 for setting max vcpu number. If a few users want to use their gic-version=2, they need to explicitly set the max vcpu to 8 in the toml configuration file.

Reply:
Thanks for the information @justin-he.